Sources/SwiftParser/Parser.swift

```diff
@@ -629,3 +658,69 @@ extension Parser {
     )
   }
 }
+
+// MARK: Incremental Parsing
```

Comment:
What do you think about moving all of the incremental-parsing functions to its own file? It just keeps them nicely grouped and maybe we can even mark some functions as `fileprivate`.

Reply:
I think that's great. And I also make `IncrementalParseLookup.lookup` an `fileprivate` function.